### PR TITLE
Experimental: Add JS tests to GitHub actions

### PR DESCRIPTION
**Pull request by duncanjbrown**
Experiment following #1844

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Seems to work. Pushing another commit to make a test fail...

**Comment by duncanjbrown** [comment]
It failed 🎉

**Comment by duncanjbrown** [comment]
There's an unrelated transient failure in the acceptance specs. Card: https://trello.com/c/jU6gucuI/1884-fix-transient-spec-failure

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -225,3 +225,72 @@ jobs:
...
+      postgres:
+        image: 'postgres:11.5'
+        ports:
+          - '5432:5432'
+        options: '--health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5'
+      redis:
+        image: redis
+        ports:
+          - '6379:6379'
+        options: '--entrypoint redis-server'
```
Comment (tvararu) [inline .github/workflows/tests.yml:243]: Don't think these are needed to run JS tests 🤔

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -225,3 +225,72 @@ jobs:
...
+          PGHOST: localhost
+          PGUSER: postgres
+          RAILS_ENV: test
+        run: |
+          yarn install
+          gem install bundler
+          bundle config path vendor/bundle
+          bundle install --jobs 4 --retry 3
+          bin/rails db:setup
+          RAILS_ENV=test bin/rails webpacker:compile
```
Comment (tvararu) [inline .github/workflows/tests.yml:292]: None of these commands and their Ruby version prerequisites should be necessary to run JS tests 🤔

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -225,3 +225,72 @@ jobs:
...
+
+      - name: 'Run tests'
+        run: |
+          npx jest
```
Comment (tvararu) [inline .github/workflows/tests.yml:296]: The `npx` equivalent for `yarn` is simply `yarn`:

```suggestion
          yarn jest
```

File: .github/workflows/tests.yml (hunk 2 above)
Comment (duncanjbrown) [reply .github/workflows/tests.yml:292]: We need to run Webpacker, which depends on Rails, which depends on... everything

File: .github/workflows/tests.yml (hunk 3 above)
Comment (duncanjbrown) [reply .github/workflows/tests.yml:296]: Thanks — forgot which JS ecosystem I was in 😂

File: .github/workflows/tests.yml (hunk 2 above)
Comment (tvararu) [reply .github/workflows/tests.yml]: Turns out we don't; we need webpacker for the JavaScript that we want to run in the browser. Jest however runs in Node, so we don’t need to compile or do anything to the code, it just runs natively.

**Pull request merged**